src/lib/layers/editable-polygons-layer.js

```diff
@@ -1,148 +1,225 @@
-// @flow
-import { point, polygon, multiPolygon } from '@turf/helpers';
+/* eslint-env browser */
 
-import LayerMouseEvent from '../layer-mouse-event';
-import Feature from '../feature';
+import { CompositeLayer, GeoJsonLayer, ScatterplotLayer } from 'deck.gl';
+import { immutablyReplaceCoordinate } from '../geojson';
 
-import PolygonsLayer from './polygons-layer';
-import EditableJunctionsLayer from './editable-junctions-layer';
+export default class EditablePolygonsLayer extends CompositeLayer {
+  static layerName = 'EditablePolygonsLayer';
 
-function deepCopyCoords(c) {
-  if (Array.isArray(c)) {
-    return c.map(deepCopyCoords);
+  renderLayers() {
+    let layers = [
+      new GeoJsonLayer({
```

Comment:
There is a prop forwarding function on `CompositeLayer` that forwards all base props, update triggers etc.

Reply:
[`getSublayerProps`](http://uber.github.io/deck.gl/#/documentation/deckgl-api-reference/layers/composite-layer?section=methods)

Maybe search for it in the code base to see examples of how it is used?

Reply:
i tried getting that to work, but am having issues. i'll refactor in a future PR